src/ScenarioManagerTreeList/ScenarioManagerTreeList.js

```diff
@@ -17,15 +17,42 @@ const ScenarioManagerTreeList = (props) => {
   const {
     datasets,
     scenarios,
-    moveScenario
+    deleteScenario,
+    moveScenario,
+    userId
   } = props;
 
   // Memoize the full scenarios tree in a ReactSortableTree-compatible format
-  const rstScenarios = useMemo(() => formatScenariosToRSTList(datasets, scenarios), [datasets, scenarios]);
+  const expandedNodes = useRef([]);
+  const rstScenarios = useMemo(() => formatScenariosToRSTList(scenarios), [datasets, scenarios]);
 
   const [searchText, setSearchText] = useState('');
   const [treeData, setTreeData] = useState(rstScenarios);
 
+  // On scenarios list update, re-apply current filter on the new list of scenarios
+  useEffect(() => {
+    filterScenarios(searchText);
+  }, [scenarios]);
+
+  function formatScenariosToRSTList (treeScenarios) {
+    const rstScenarios = treeScenarios.map((scenario) => {
+      const showDeleteIcon = userId && scenario.ownerId === userId;
```

Comment:
If you change the requirement on userId, this line can be `scenario.ownerId === userId`

Reply:
Done :+1: